### PR TITLE
feat(compose-up): :sparkles: add configurable environment variables

### PR DESCRIPTION
**Pull request by LoicFerrot**
### Please check if the PR fulfills these requirements
- [ ] Tests for the changes have been added (for bug fixes/features)
- [x] This compose-up workflow ran successfully with mantis-streamer

### What kind of change does this PR introduce?
Feature / fix

### What is the current behavior?
The environment variables available to the `docker compose up` command are defined in this repo and are not extensible from other repositories. 

### What is the new behavior?
It is possible to add environment variables from the repositories using this action.

### Does this PR introduce a breaking change?
No, only a new optional parameter is defined.

### Other information
(E.g. why is this PR needed, etc.)
This PR is needed mainly to be able to set the environment variables necessary for the new docker-compose in mantis-streamer. See [this PR](https://github.com/mantis-tec/mantis-streamer-ros2/pull/22).

The step `Setup environment variables` is necessary, because no environment variables can be directly passed from the calling workflow (see the [limitations of the reusable workflows](https://docs.github.com/en/actions/using-workflows/reusing-workflows#limitations)).
The workaround used there is to use an input to the reusable workflow.
The use of JSON logic is needed because the input can only be of type string, it cannot be a mapping.

### CODE REVIEW

File: .github/workflows/compose-up.yml [hunk 1]
```diff
@@ -43,11 +47,17 @@ jobs:
...
+              "LOG_LEVEL": "debug",
+              "IMAGE_TAG": "${{ fromJSON(steps.meta.outputs.json).tags[0] }}"
+            }
+        run: echo '${{ env.env_vars }} ${{ inputs.compose-up-env-variables }}' | jq -r -s 'add | to_entries | map("\(.key)=\(.value | tostring)") | .[]' >> $GITHUB_ENV
+
       - name: Compose Up
-        env:
-          NAMESPACES: stand1,stand2,stand3,stand4
-          LOG_LEVEL: debug
-          IMAGE_TAG: ${{ fromJSON(steps.meta.outputs.json).tags[0] }}
```
Comment (kevinanschau) [inline .github/workflows/compose-up.yml:50]: Is it necessary to _merge_ the two different sets of environment variables? Isn't it possible to just add the input env variables to the `GITHUB_ENV` variable and then keep the removed lines?

File: .github/workflows/compose-up.yml [hunk 2]
```diff
@@ -7,6 +7,10 @@ on:
         required: False
         type: string
         default: staging
+      compose-up-env-variables:
```
Comment (kevinanschau) [inline .github/workflows/compose-up.yml:10]: Nitpick: This input could be used to pass environment variables to any step in the reusable workflow. Hence, IMO, maybe rename it to `env`.

File: .github/workflows/compose-up.yml [hunk 3]
```diff
@@ -7,6 +7,10 @@ on:
...
+        required: False
+        description: A Json string containing the environment variables to use in the Compose Up step.
```
Comment (kevinanschau) [inline .github/workflows/compose-up.yml:12]: Nitpick: If you change the input's name, remove the compose specification in the description.

**Review by kevinanschau — CHANGES_REQUESTED** [review]
Thanks for this feature. Please see my comments.

File: .github/workflows/compose-up.yml (hunk 1 above)
Comment (LoicFerrot) [reply .github/workflows/compose-up.yml:50]: Yes, it works and simplifies a bit the `jq` expression :+1:

**Review by kevinanschau — APPROVED** [review]
Thanks for the changes. LGTM!

**Pull request merged**